sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient_helpers.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+
+namespace Azure.Messaging.WebPubSub
+{
+    /// <summary>
+    /// Azure Web PubSub Service Client.
+    /// </summary>
+    public partial class WebPubSubServiceClient
+    {
+        private const string EndpointPropertyName = "Endpoint";
+        private const string AccessKeyPropertyName = "AccessKey";
+        private const string PortPropertyName = "Port";
+        private static readonly char[] KeyValueSeparator = { '=' };
+        private static readonly char[] PropertySeparator = { ';' };
+
+        /// <summary>
+        /// Creates a URI with authentication token.
+        /// </summary>
+        /// <returns></returns>
+        public virtual Uri GetClientAccessUri(string userId = default, string[] roles = default, TimeSpan expireAfter = default)
+        {
+            if (expireAfter == default)
+            {
+                expireAfter = TimeSpan.FromHours(1);
+            }
+
+            List<Claim> claims = new List<Claim>();
+            if (userId != default)
+            {
+                var subject = new Claim("sub", userId);
+                claims.Add(subject);
+            }
+            if (roles != default && roles.Length > 0)
+            {
+                var jsonArray = BinaryData.FromObjectAsJson(roles).ToString();
+                var role = new Claim("role", jsonArray);
+                claims.Add(role);
+            }
+
+            var endpoint = _endpoint.ToString().EndsWith("/", StringComparison.Ordinal) ? _endpoint.ToString() : _endpoint + "/";
+            var audience = $"{endpoint}client/hubs/{hub}";
+
+            var token = WebPubSubAuthenticationPolicy.GenerateAccessToken(audience, claims: claims, _credential, expireAfter);
+
+            var clientEndpoint = new UriBuilder(endpoint);
+            clientEndpoint.Scheme = "wss";
+            var uriString = $"{clientEndpoint}client/hubs/{hub}?access_token={token}";
+
+            return new Uri(uriString);
```

Comment:
Why not use the builder instead of going back and forth?
```suggestion
            clientEndpoint.Path += $"client/hubs/{hub}";
            clientEndpoint.Query = $"access_token={token}";
            
            return clientEndpoint.Uri;
```